### ISSUE TITLE
[Feature proposal] Apply adjust_contrast transformation for grayscale tensors

### ISSUE THREAD

**Issue by ukky17** [pytorch/vision#3670]
## Motivation

Currently, `adjust_contrast` function in `torchvision.transformations.functional_tensor` only works for 3-channel (=RGB) tensors. However, I would like to change the contrast of a 1-channel (= grayscale) tensor.

## Pitch

Just like the 3-channel case, the desired function changes the image contrast. I.e., `contrast_factor = 0` gives a uniform gray image, 1 gives the original image while 2 increases the contrast by a factor of 2.

## Additional context

This feature can be implemented by slightly modifying the `adjust_contrast` function in ``torchvision.transformations.functional_tensor` like this;

```
def adjust_contrast(img: Tensor, contrast_factor: float) -> Tensor:
    if contrast_factor < 0:
        raise ValueError('contrast_factor ({}) is not non-negative.'.format(contrast_factor))

    _assert_image_tensor(img)

    _assert_channels(img, [1, 3])

    dtype = img.dtype if torch.is_floating_point(img) else torch.float32

    num_channels = _get_image_num_channels(img)
    if num_channels == 1:
        gray_img = img.to(dtype)
    elif num_channels == 3:
        gray_img = rgb_to_grayscale(img).to(dtype)

    mean = torch.mean(gray_img, dim=(-3, -2, -1), keepdim=True)

    return _blend(img, mean, contrast_factor)
```

**Comment by datumbox** [comment]
Thanks for the proposal. 

I added it in the list of known limitations being tracked at https://github.com/pytorch/vision/issues/3224

As a temporary workaround, convert the image to RGB by copying the same colours in all channels, apply the transforms you want and then covert it back to grayscale using [rgb_to_grayscale()](https://github.com/pytorch/vision/blob/9778d2675ed33627c30fe2537911f5a4128433bd/torchvision/transforms/functional.py#L1116). Suboptimal but can temporarily help you workaround the problem until the limitations are lifted.

**Comment by voldemortX** [comment]
@datumbox I think this kind of relate to the same "Tensor images do not have meta-data" problem. Since we assume RGB inputs for all tensor transforms. #3219

The problem is I'm not sure if we can make a good guess about the image mode on dtype & channel numbers, or if we can add an input variable about image mode (that would introduce inconsistency between the two backends). On the other hand, adding the meta-data on tensor images would be too big a change to the current framework. But maybe torchvision could need meta-data in the future, as there is native image I/O functions being provided in `torchvision.io` ?

**Comment by datumbox** [comment]
Yes it's related. I've added it on your list on the ticket just to be safe.

One workaround in this case would be to convert the image to RGB, apply the transforms of TorchVision and convert back to grayscale just before modeling. This should be solved more naturally once we support a better solution in the transforms for different image types.

**Comment by ukky17** [comment]
Thank you for the comments!

Yes, temporarily converting the image to RGB is a workaround, but I believe that slightly changing `functional_tensor.py` (like the code snippet in my first comment) would be a natural solution. Let me know if it's OK to send a PR.
